app/scripts/controllers/rif/rns/resolver/index.js

```diff
@@ -47,4 +50,56 @@ export default class RnsResolver extends RnsJsDelegate {
         }).catch(error => reject(error));
     });
   }
+  
+  /**
+   * Gets all details of a given domain name
+   * @param domainName the domain name to check (without resolver).
+   * @returns {Promise<boolean>} true if it can get all the details correctly, false otherwise.
+   */
+  getDomainDetails(domainName) {
+    const domainNameResolver = domainName + ".rsk";
+    return new Promise((resolve, reject) => {
+      this.getExpirationRemaining(domainNameResolver).then(remainingDays => { 
+        //Here i have the expiration in remainingDays
+        this.getOwner(domainNameResolver)
+        .then(ownerAddress => {
+          //Here i have the owner address in ownerAddress
+          this.getDomainAddress(domainNameResolver).then(domainAddress => {
+            //Here i have the domain address in domainAddress
+            resolve(new DomainDetails(domainNameResolver, domainAddress, '0xabcd', remainingDays, false, ownerAddress));
+          }).catch(error => reject(error));        
+        }).catch(error => reject(error));        
+      }).catch(error => reject(error));
+    });
+  }
+
+  /**
+   * Gets the expiration time in days of a given domain name
+   * @param domainName the domain name to check (without resolver).
+   * @returns {Promise<boolean>} Days remaining (int), an error otherwise
+   */
+  getExpirationRemaining(domainName) {  
+    return new Promise((resolve, reject) => {
+      const label = domainName.split('.')[0];
+      const hash = `0x${sha3(label)}`;
+      this.rskOwnerContractInstance.expirationTime(hash, (error, result) => {
```

Comment:
We need to make a standard on contract invokations.

Take a look at `rns-delegate ` `call ` method https://github.com/anarancio/nifty-wallet/pull/8/files#diff-fd3f70ef0d29a042399ae36fc5301309

Lets discuss this with @JONAF2103 at tomorrow daily

Reply:
marcosmartinez7 17 hours ago Collaborator
We need to make a standard on contract invokations.

Take a look at rns-delegate call method https://github.com/anarancio/nifty-wallet/pull/8/files#diff-fd3f70ef0d29a042399ae36fc5301309

Lets discuss this with @JONAF2103 at tomorrow daily

Reply:
It have changed, check the new code please. Its implemented with .call from @JONAF2103